modules/sync/syncs/httpsynchronization.cpp

```diff
@@ -219,33 +227,57 @@ bool HttpSynchronization::trySyncFromUrl(std::string listUrl) {
             ghoul::filesystem::FileSystem::PathSeparator +
             filename + TempSuffix;
 
+        if (sizeData.find(line) != sizeData.end()) {
+            LWARNING(
+                fmt::format("{}: Duplicate entries: {}", _identifier, line)
+            );
+            continue;
+        }
+
         downloads.push_back(std::make_unique<AsyncHttpFileDownload>(
             line, fileDestination, HttpFileDownload::Overwrite::Yes));
 
         auto& fileDownload = downloads.back();
 
-        ++nDownloads;
+        sizeData[line] = {
+            false,
+            0,
+            0,
+        };
 
         fileDownload->onProgress(
-            [this, line, &fileSizes, &fileSizeMutex,
-             &startedAllDownloads, &nDownloads](HttpRequest::Progress p)
+            [this, line, &sizeData, &sizeDataMutex,
+             &startedAllDownloads](HttpRequest::Progress p)
         {
-            if (p.totalBytesKnown) {
-                std::lock_guard<std::mutex> guard(fileSizeMutex);
-                fileSizes[line] = p.totalBytes;
-
-                if (!_nTotalBytesKnown && startedAllDownloads &&
-                    fileSizes.size() == nDownloads)
-                {
-                    _nTotalBytesKnown = true;
-                    _nTotalBytes = std::accumulate(
-                        fileSizes.begin(),
-                        fileSizes.end(),
-                        size_t(0),
-                        [](size_t a, auto b) { return a + b.second; }
-                    );
-                }
+            if (!p.totalBytesKnown || !startedAllDownloads) {
+                return !_shouldCancel;
             }
+
+            std::lock_guard<std::mutex> guard(sizeDataMutex);
+
+            sizeData[line] = {
+                p.totalBytesKnown,
+                p.totalBytes,
+                p.downloadedBytes,
+            };
+
+            SizeData size = std::accumulate(
+                sizeData.begin(),
+                sizeData.end(),
+                SizeData{ true, 0, 0 },
+                [](const SizeData& a, const auto& b) -> SizeData {
```

Comment:
The auto is not necessary as it is the type of `sizeData`, which is `std::pair<const std::string, SizeData>`